changelogs/fragments/play-argument-spec-validation.yml

```diff
@@ -0,0 +1,7 @@
+minor_changes:
+- >-
+  Add tech preview play argument spec validation, which can be
+  enabled by setting the play keyword ``validate_argspec`` to ``True``.
+  This will load a file matching the pattern <playbook_name>.meta.yml,
+  which should contain a top dictionary "argument_specs", containing
+  the play name, containing an "options" dictionary.
```

Comment:
IIRC, for role argspec validation, the existence of an argspec file means we automatically do the validation. I think `include_role` and `import_role` have options to disable. Perhaps we follow the same pattern for consistency sake? If an argspec exists and a play name matches, always do the validation. The new option could then be used to explicitly disable.

Reply:
The goal was to make it explicitly opt in, to also catch the case of a play being renamed, which is potentially more likely than a role file being renamed. If it were explicitly enabled, you would know when the argspec didn't match because of getting an error.

On the flip side, renaming the play would just result in no argspec validation happening.

I think we want to stick with an explicit enable.

Reply:
@sivel That makes sense. The role task entrypoints would need to be renamed, just renaming the role wouldn't break the role argument spec validation. Whereas implicit play argument spec validation would break by renaming the playbook or play, or even by providing different extra vars if play names are dynamic. I think I like the simplicity of requiring opt-in.

What about allowing explicit opt-in to enforce validation, but implicitly validate arguments if the keyword is omitted and a play argument spec exists? Maybe this would cause more confusion/risk than the convenience it provides. :thinking: